### DISCUSSION TITLE
ContractFunctionExecutionError: The request took too long to respond.

### DISCUSSION THREAD

**Discussion by dixitt5** [wevm/wagmi#2768] [Q&A] [answered]
I'm using the writecontract function of wagmi/core, and when i call the function, it returns back with an error after some time which is : 
`client.js:1 ContractFunctionExecutionError: The request took too long to respond.

URL: https://matic-mumbai.chainstacklabs.com`

I've tried the signaturemessage function too to check if i've setup wagmi correctly and it's working perfectly.

**Comment by talha-amin** [comment]
Any leads on how to fix?

**Reply by dixitt5** [reply]
@talha-amin No, I haven't been able to find a solution.

**Reply by talha-amin** [reply, marked as answer]
Here i found it 

const { chains,publicClient } = configureChains(
  [polygonMumbai],
  [
    jsonRpcProvider({
      rpc: (chain) => ({
        http: `https://polygon-mumbai-bor.publicnode.com`,
      }),
    }),
  ]
);

**Comment by NishadVadgama** [comment]
Thanks @talha-amin for the answer.

For me following worked.

```typescript
const { chains, publicClient, webSocketPublicClient } = configureChains(
  [mainnet, polygon, avalanche, polygonMumbai, avalancheFuji],
  [
    jsonRpcProvider({
      rpc: (chain) => ({
        http: `https://rpc-mumbai.maticvigil.com`,
      }),
    }),
    publicProvider(),
  ]
)
```

**Reply by phongkhuu115** [reply]
@NishadVadgama May I ask where you put the code, and use the returns value?
